twisted/internet/_sslverify.py

```diff
@@ -1645,11 +1648,11 @@ def _verifyCallback(conn, cert, errno, depth, preverify_ok):
             name = "%s-%d" % (reflect.qual(self.__class__), _sessionCounter())
             sessionName = md5(networkString(name)).hexdigest()
 
-            ctx.set_session_id(sessionName)
+            ctx.set_session_id(sessionName.encode('ascii'))
```

Comment:
As far as I can see this method doesn't have any docstring. Also sessionName is not a passed argument rather calculated using call to hexdigest(), which returns a native `str` object on both python2 and 3. https://docs.python.org/3/library/hashlib.html, https://docs.python.org/2/library/hashlib.html